### ISSUE TITLE
Support methods defined by macros

### ISSUE THREAD

**Issue by daboross** [intellij-rust/intellij-rust#4153]
I'm not 100% sure whether this should be working - but if it would be possible, I would love to have this as a feature!

## Environment

* **IntelliJ Rust plugin version:** v0.2.101.2129-191
* **Rust toolchain version:** rustc 1.38.0-nightly (07e0c3651 2019-07-16)
* **IDE name and version:**

```
IntelliJ IDEA 2019.1.3 (Ultimate Edition)
Build #IU-191.7479.19, built on May 27, 2019
JRE: 1.8.0_202-release-1483-b58 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 5.1.15-1-default
```

* **Operating system:** OpenSUSE Tumbleweed

## Description

When a `macro_rules!` macro defines a method on an existing type, that method is not registered in Intellij-Rust.

## Related issues

- https://github.com/intellij-rust/intellij-rust/issues/2458 (closed)
- https://github.com/intellij-rust/intellij-rust/issues/2787 (different in that type is defined inside the macro)

## Steps to reproduce

Create a new library Rust project, then replace the lib file with the following code:

```rust
macro_rules! define_methods {
    ($($method:ident()),* $(,)*) => (
        impl MyStruct {
            $(
                pub fn $method(&self) { }
            )*
        }
    )
}

struct MyStruct;

define_methods! {
    method_1(),
    m2(),
}

#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        let x = crate::MyStruct;
        x.method_1();
    }
}

```

On a new line after `x.method_1();`, attempt to autocomplete `x.m<ctrl+space>`. Neither `method_1` nor `m2` shows up.

Hope that's enough information! If there's anything I can add, feel free to ask.

Thanks!

**Comment by Undin** [comment]
@daboross Yeah, our default macro expansion engine can't handle methods generated by macro. But we implemented new macro expansion engine to fix this issue (it was implemented in #2458)
It should solve your issue.
You can enable it via `Preferences | Languages & Frameworks | Rust | Expand declarative macros option`.
![image](https://user-images.githubusercontent.com/2539310/58436688-4f633c80-80cf-11e9-9bcf-0f81bf30f46b.png)
Note, this engine is experimental and not stable yet so don't hesitate to share your feedback. It can help us to polish this feature and make it default. More about new macro engine can be found in #3628

**Comment by daboross** [comment]
Oh, thanks!

For some reason I thought I had enabled that, but I realize I hadn't. With it enabled, this does auto-complete.
